`Error in the instrumented process |> RdFault: SpringBeforeTestMethodPhase` on test generation for `OwnerController` in PetClinic

**Description**
Errors happens on integration test generation for controller from Pet Clinic Spring project.


**To Reproduce**

1. Install UnitTestBot plugin built from main in IntelliJ IDEA
2. Clone and open `spring-petclinic` project - https://github.com/spring-projects/spring-petclinic/
3. Generate tests for org.springframework.samples.petclinic.owner.OwnerController:
   * Configuration: `PetClinicApplication`
   * Type: `Integration tests` 

**Expected behavior**
Tests are generated


**Actual behavior**
No tests are generated. Exceptions in `utbot-engine-current.log`

<details>
<summary>utbot-engine-current.log</summary>
15:46:52.533 | ERROR | TestCaseGenerator         | UtError occurred
org.utbot.instrumentation.util.InstrumentedProcessError: Error in the instrumented process |> com.jetbrains.rd.util.reactive.RdFault: SpringBeforeTestMethodPhase, reason: org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseError: SpringBeforeTestMethodPhase
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseFailingOnAnyException.wrapError(ExecutionPhase.kt:35)
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseKt.start(ExecutionPhase.kt:30)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController.executePhaseWithoutTimeout(PhasesController.kt:75)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:46)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.SimpleUtExecutionInstrumentation.invoke(SimpleUtExecutionInstrumentation.kt:62)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$DefaultImpls.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:131)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:128)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Cannot start new transaction without ending existing transaction
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:204)
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:288)
	at org.utbot.spring.SpringApiImpl.beforeTestMethod(SpringApiImpl.kt:149)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController$executePhaseWithoutTimeout$1$1.invoke(PhasesController.kt:77)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:91)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:87)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)

	at com.jetbrains.rd.framework.RdTaskResult$Companion.read(TaskInterfaces.kt:30)
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask.onWireReceived(RdTask.kt:106)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.queue$execute(RdTask.kt:280)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.access$queue$execute(RdTask.kt:269)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1$queue$1.invokeSuspend(RdTask.kt:289)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.utbot.rd.UtRdUtilKt.startBlocking(UtRdUtil.kt:31)
	at org.utbot.rd.RdSettingsContainer$settingFor$1$1.getValue(RdSettingsContainer.kt:33)
	at org.utbot.framework.UtSettings.getUseAssembleModelGenerator(UtSettings.kt:205)
	at org.utbot.framework.assemble.AssembleModelGenerator.createAssembleModels(AssembleModelGenerator.kt:115)
	at org.utbot.framework.util.UtConcreteExecutionResultUtilsKt.convertToAssemble(UtConcreteExecutionResultUtils.kt:40)
	at org.utbot.engine.UtBotSymbolicEngineKt.executeConcretely(UtBotSymbolicEngine.kt:798)
	at org.utbot.engine.UtBotSymbolicEngineKt.access$executeConcretely(UtBotSymbolicEngine.kt:1)
	at org.utbot.engine.UtBotSymbolicEngineKt$executeConcretely$1.invokeSuspend(UtBotSymbolicEngine.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.utbot.common.ConcurrencyKt.runBlockingWithCancellationPredicate(Concurrency.kt:38)
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:196)
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:195)
	at org.utbot.common.ConcurrencyKt.runIgnoringCancellationException(Concurrency.kt:47)
	at org.utbot.framework.plugin.api.TestCaseGenerator.generate(TestCaseGenerator.kt:195)
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:135)
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:120)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

	at org.utbot.instrumentation.ConcreteExecutor.withProcess(ConcreteExecutor.kt:213) ~[utbot-instrumentation-2023.8.4997.jar:?]
	at org.utbot.instrumentation.ConcreteExecutor$withProcess$1.invokeSuspend(ConcreteExecutor.kt) ~[utbot-instrumentation-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.rd.UtRdUtilKt.startBlocking(UtRdUtil.kt:31) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.RdSettingsContainer$settingFor$1$1.getValue(RdSettingsContainer.kt:33) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.framework.UtSettings.getUseAssembleModelGenerator(UtSettings.kt:205) ~[utbot-framework-api-2023.8.4997.jar:?]
	at org.utbot.framework.assemble.AssembleModelGenerator.createAssembleModels(AssembleModelGenerator.kt:115) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.util.UtConcreteExecutionResultUtilsKt.convertToAssemble(UtConcreteExecutionResultUtils.kt:40) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.executeConcretely(UtBotSymbolicEngine.kt:798) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.access$executeConcretely(UtBotSymbolicEngine.kt:1) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt$executeConcretely$1.invokeSuspend(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.rd.UtRdUtilKt.startBlocking(UtRdUtil.kt:31) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.RdSettingsContainer$settingFor$1$1.getValue(RdSettingsContainer.kt:33) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.framework.UtSettings.getConcreteExecutionDefaultTimeoutInInstrumentedProcessMillis(UtSettings.kt:269) ~[utbot-framework-api-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$2$1.invokeSuspend(UtBotSymbolicEngine.kt:486) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$2$1.invoke(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$2$1.invoke(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invokeSuspend(JavaLanguage.kt:129) ~[utbot-java-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invoke(JavaLanguage.kt) ~[utbot-java-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invoke(JavaLanguage.kt) ~[utbot-java-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.BaseFuzzing.handle(Providers.kt:56) ~[utbot-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.FuzzingApi.fuzz(Api.kt:359) ~[utbot-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.FuzzingApi.access$fuzz(Api.kt:1) ~[utbot-fuzzing-2023.8.4997.jar:?]
	at org.utbot.fuzzing.FuzzingApi$fuzz$2.invokeSuspend(Api.kt) ~[utbot-fuzzing-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.rd.UtRdUtilKt.startBlocking(UtRdUtil.kt:31) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.RdSettingsContainer$settingFor$1$1.getValue(RdSettingsContainer.kt:33) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.framework.UtSettings.getUseAssembleModelGenerator(UtSettings.kt:205) ~[utbot-framework-api-2023.8.4997.jar:?]
	at org.utbot.framework.assemble.AssembleModelGenerator.createAssembleModels(AssembleModelGenerator.kt:115) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.util.UtConcreteExecutionResultUtilsKt.convertToAssemble(UtConcreteExecutionResultUtils.kt:40) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.executeConcretely(UtBotSymbolicEngine.kt:798) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.access$executeConcretely(UtBotSymbolicEngine.kt:1) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt$executeConcretely$1.invokeSuspend(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.common.ConcurrencyKt.runBlockingWithCancellationPredicate(Concurrency.kt:38) ~[utbot-core-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:196) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:195) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.common.ConcurrencyKt.runIgnoringCancellationException(Concurrency.kt:47) ~[utbot-core-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator.generate(TestCaseGenerator.kt:195) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:135) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:120) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114) ~[utbot-rd-2023.8.4997.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41) ~[rd-core-2023.1.2.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: com.jetbrains.rd.util.reactive.RdFault: SpringBeforeTestMethodPhase, reason: org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseError: SpringBeforeTestMethodPhase
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseFailingOnAnyException.wrapError(ExecutionPhase.kt:35)
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseKt.start(ExecutionPhase.kt:30)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController.executePhaseWithoutTimeout(PhasesController.kt:75)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:46)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.SimpleUtExecutionInstrumentation.invoke(SimpleUtExecutionInstrumentation.kt:62)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$DefaultImpls.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:131)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:128)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Cannot start new transaction without ending existing transaction
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:204)
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:288)
	at org.utbot.spring.SpringApiImpl.beforeTestMethod(SpringApiImpl.kt:149)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController$executePhaseWithoutTimeout$1$1.invoke(PhasesController.kt:77)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:91)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:87)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)

	at com.jetbrains.rd.framework.RdTaskResult$Companion.read(TaskInterfaces.kt:30) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask.onWireReceived(RdTask.kt:106) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.queue$execute(RdTask.kt:280) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.access$queue$execute(RdTask.kt:269) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1$queue$1.invokeSuspend(RdTask.kt:289) ~[rd-framework-2023.1.2.jar:?]
	... 46 more
15:46:54.340 | ERROR | ConcreteExecutor          | executeAsync, response(ERROR)
org.utbot.instrumentation.util.InstrumentedProcessError: Error in the instrumented process |> com.jetbrains.rd.util.reactive.RdFault: SpringBeforeTestMethodPhase, reason: org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseError: SpringBeforeTestMethodPhase
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseFailingOnAnyException.wrapError(ExecutionPhase.kt:35)
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseKt.start(ExecutionPhase.kt:30)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController.executePhaseWithoutTimeout(PhasesController.kt:75)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:46)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.SimpleUtExecutionInstrumentation.invoke(SimpleUtExecutionInstrumentation.kt:62)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$DefaultImpls.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:131)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:128)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Cannot start new transaction without ending existing transaction
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:204)
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:288)
	at org.utbot.spring.SpringApiImpl.beforeTestMethod(SpringApiImpl.kt:149)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController$executePhaseWithoutTimeout$1$1.invoke(PhasesController.kt:77)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:91)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:87)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)

	at com.jetbrains.rd.framework.RdTaskResult$Companion.read(TaskInterfaces.kt:30)
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask.onWireReceived(RdTask.kt:106)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.queue$execute(RdTask.kt:280)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.access$queue$execute(RdTask.kt:269)
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1$queue$1.invokeSuspend(RdTask.kt:289)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at org.utbot.common.ConcurrencyKt.runBlockingWithCancellationPredicate(Concurrency.kt:38)
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:196)
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:195)
	at org.utbot.common.ConcurrencyKt.runIgnoringCancellationException(Concurrency.kt:47)
	at org.utbot.framework.plugin.api.TestCaseGenerator.generate(TestCaseGenerator.kt:195)
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:135)
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:120)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

	at org.utbot.instrumentation.ConcreteExecutor.withProcess(ConcreteExecutor.kt:213) ~[utbot-instrumentation-2023.8.4997.jar:?]
	at org.utbot.instrumentation.ConcreteExecutor$withProcess$1.invokeSuspend(ConcreteExecutor.kt) ~[utbot-instrumentation-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.rd.UtRdUtilKt.startBlocking(UtRdUtil.kt:31) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.RdSettingsContainer$settingFor$1$1.getValue(RdSettingsContainer.kt:33) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.framework.UtSettings.getUseAssembleModelGenerator(UtSettings.kt:205) ~[utbot-framework-api-2023.8.4997.jar:?]
	at org.utbot.framework.assemble.AssembleModelGenerator.createAssembleModels(AssembleModelGenerator.kt:115) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.util.UtConcreteExecutionResultUtilsKt.convertToAssemble(UtConcreteExecutionResultUtils.kt:40) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.executeConcretely(UtBotSymbolicEngine.kt:798) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt.access$executeConcretely(UtBotSymbolicEngine.kt:1) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.engine.UtBotSymbolicEngineKt$executeConcretely$1.invokeSuspend(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.8.4997.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.common.ConcurrencyKt.runBlockingWithCancellationPredicate(Concurrency.kt:38) ~[utbot-core-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:196) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:195) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.common.ConcurrencyKt.runIgnoringCancellationException(Concurrency.kt:47) ~[utbot-core-2023.8.4997.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator.generate(TestCaseGenerator.kt:195) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:135) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$4.invoke(EngineProcessMain.kt:120) ~[utbot-framework-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88) ~[utbot-rd-2023.8.4997.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114) ~[utbot-rd-2023.8.4997.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41) ~[rd-core-2023.1.2.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: com.jetbrains.rd.util.reactive.RdFault: SpringBeforeTestMethodPhase, reason: org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseError: SpringBeforeTestMethodPhase
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseFailingOnAnyException.wrapError(ExecutionPhase.kt:35)
	at org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhaseKt.start(ExecutionPhase.kt:30)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController.executePhaseWithoutTimeout(PhasesController.kt:75)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:46)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$invoke$1.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:71)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.SimpleUtExecutionInstrumentation.invoke(SimpleUtExecutionInstrumentation.kt:62)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:70)
	at org.utbot.instrumentation.instrumentation.execution.UtExecutionInstrumentation$DefaultImpls.invoke(UtExecutionInstrumentation.kt:45)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation.invoke(SpringUtExecutionInstrumentation.kt:31)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:131)
	at org.utbot.instrumentation.process.InstrumentedProcessMainKt$setup$2.invoke(InstrumentedProcessMain.kt:128)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115)
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88)
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182)
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56)
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54)
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Cannot start new transaction without ending existing transaction
	at org.springframework.util.Assert.state(Assert.java:76)
	at org.springframework.test.context.transaction.TransactionalTestExecutionListener.beforeTestMethod(TransactionalTestExecutionListener.java:204)
	at org.springframework.test.context.TestContextManager.beforeTestMethod(TestContextManager.java:288)
	at org.utbot.spring.SpringApiImpl.beforeTestMethod(SpringApiImpl.kt:149)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.spring.SpringUtExecutionInstrumentation$invoke$2$1$1.invoke(SpringUtExecutionInstrumentation.kt:77)
	at org.utbot.instrumentation.instrumentation.execution.phases.PhasesController$executePhaseWithoutTimeout$1$1.invoke(PhasesController.kt:77)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:91)
	at org.utbot.common.ThreadBasedExecutor$ensureThreadIsAlive$1.invoke(ThreadUtil.kt:87)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)

	at com.jetbrains.rd.framework.RdTaskResult$Companion.read(TaskInterfaces.kt:30) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.CallSiteWiredRdTask.onWireReceived(RdTask.kt:106) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.queue$execute(RdTask.kt:280) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1.access$queue$execute(RdTask.kt:269) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall$createResponseScheduler$1$queue$1.invokeSuspend(RdTask.kt:289) ~[rd-framework-2023.1.2.jar:?]
	... 30 more
</details>

--- a/utbot-core/src/main/kotlin/org/utbot/common/ThreadUtil.kt
+++ b/utbot-core/src/main/kotlin/org/utbot/common/ThreadUtil.kt
@@ -1,5 +1,6 @@
 package org.utbot.common
 
+import java.util.WeakHashMap
 import java.util.concurrent.ArrayBlockingQueue
 import java.util.concurrent.TimeUnit
 import kotlin.concurrent.thread
@@ -24,11 +25,15 @@ class ThreadBasedExecutor {
         val threadLocal by threadLocalLazy { ThreadBasedExecutor() }
     }
 
+    // there's no `WeakHashSet`, so we use `WeakHashMap` with dummy values
+    private val timedOutThreads = WeakHashMap<Thread, Unit>()
     private var thread: Thread? = null
 
     private var requestQueue = ArrayBlockingQueue<() -> Any?>(1)
     private var responseQueue = ArrayBlockingQueue<Result<Any?>>(1)
 
+    fun isCurrentThreadTimedOut(): Boolean =
+        Thread.currentThread() in timedOutThreads
 
     /**
      * Invoke [action] with timeout.
@@ -59,6 +64,7 @@ class ThreadBasedExecutor {
         if (res == null) {
             try {
                 val t = thread ?: return res
+                timedOutThreads[t] = Unit
                 t.interrupt()
                 t.join(10)
                 if (t.isAlive)

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/InstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/InstrumentationContext.kt
@@ -36,13 +36,11 @@ interface InstrumentationContext {
     fun findUtModelWithCompositeOriginConstructor(classId: ClassId): UtModelWithCompositeOriginConstructor?
 
     /**
-     * Called when [failedPhase] fails while being executed under timeout.
-     * This method is executed in the same thread that [failedPhase] was run in.
-     * Implementor are expected to only perform some clean up operations (e.g. rollback transactions in Spring).
-     *
-     * In case of timeout [throwable] can be [ThreadDeath], [InterruptedException], or any other [Throwable].
+     * Called when [timedOutedPhase] times out.
+     * This method is executed in the same thread that [timedOutedPhase] was run in.
+     * Implementor is expected to only perform some clean up operations (e.g. rollback transactions in Spring).
      */
-    fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable)
+    fun onPhaseTimeout(timedOutedPhase: ExecutionPhase)
 
     object MockGetter {
         data class MockContainer(private val values: List<*>) {

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/InstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/InstrumentationContext.kt
@@ -4,6 +4,7 @@ import org.utbot.framework.plugin.api.ClassId
 import org.utbot.framework.plugin.api.UtConcreteValue
 import org.utbot.framework.plugin.api.UtModel
 import org.utbot.instrumentation.instrumentation.execution.constructors.UtModelWithCompositeOriginConstructor
+import org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhase
 import java.lang.reflect.Method
 import java.util.IdentityHashMap
 import org.utbot.instrumentation.instrumentation.mock.computeKeyForMethod
@@ -33,6 +34,15 @@ interface InstrumentationContext {
      * construct models for instances of specified [class][classId].
      */
     fun findUtModelWithCompositeOriginConstructor(classId: ClassId): UtModelWithCompositeOriginConstructor?
+
+    /**
+     * Called when [failedPhase] fails while being executed under timeout.
+     * This method is executed in the same thread that [failedPhase] was run in.
+     * Implementor are expected to only perform some clean up operations (e.g. rollback transactions in Spring).
+     *
+     * In case of timeout [throwable] can be [ThreadDeath], [InterruptedException], or any other [Throwable].
+     */
+    fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable)
 
     object MockGetter {
         data class MockContainer(private val values: List<*>) {

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/SimpleInstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/SimpleInstrumentationContext.kt
@@ -6,6 +6,7 @@ import org.utbot.framework.plugin.api.UtModel
 import org.utbot.framework.plugin.api.util.jClass
 import org.utbot.instrumentation.instrumentation.execution.constructors.UtModelWithCompositeOriginConstructor
 import org.utbot.instrumentation.instrumentation.execution.constructors.javaStdLibModelWithCompositeOriginConstructors
+import org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhase
 
 /**
  * Simple instrumentation context, that is used for pure JVM projects without
@@ -21,4 +22,6 @@ class SimpleInstrumentationContext : InstrumentationContext {
 
     override fun findUtModelWithCompositeOriginConstructor(classId: ClassId): UtModelWithCompositeOriginConstructor? =
         javaStdLibModelWithCompositeOriginConstructors[classId.jClass]?.invoke()
+
+    override fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable) = Unit
 }

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/SimpleInstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/context/SimpleInstrumentationContext.kt
@@ -23,5 +23,5 @@ class SimpleInstrumentationContext : InstrumentationContext {
     override fun findUtModelWithCompositeOriginConstructor(classId: ClassId): UtModelWithCompositeOriginConstructor? =
         javaStdLibModelWithCompositeOriginConstructors[classId.jClass]?.invoke()
 
-    override fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable) = Unit
+    override fun onPhaseTimeout(timedOutedPhase: ExecutionPhase) = Unit
 }

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/phases/PhasesController.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/phases/PhasesController.kt
@@ -66,18 +66,14 @@ class PhasesController(
         val stopWatch = StopWatch()
         val context = UtContext(utContext.classLoader, stopWatch)
         val timeoutForCurrentPhase = timeout - currentlyElapsed
-        val result = ThreadBasedExecutor.threadLocal.invokeWithTimeout(timeout - currentlyElapsed, stopWatch) {
+        val executor = ThreadBasedExecutor.threadLocal
+        val result = executor.invokeWithTimeout(timeout - currentlyElapsed, stopWatch) {
             withUtContext(context) {
                 try {
                     phase.block()
-                } catch (outerThrowable: Throwable) {
-                    try {
-                        instrumentationContext.onPhaseUnderTimeoutFailure(phase, outerThrowable)
-                    } catch (innerThrowable: Throwable) {
-                        logger.error("onPhaseUnderTimeoutFailure failed", innerThrowable)
-                    } finally {
-                        throw outerThrowable
-                    }
+                } finally {
+                    if (executor.isCurrentThreadTimedOut())
+                        instrumentationContext.onPhaseTimeout(phase)
                 }
             }
         } ?: throw TimeoutException("Timeout $timeoutForCurrentPhase ms for phase ${phase.javaClass.simpleName} elapsed, controller timeout - $timeout")

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/phases/PhasesController.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/execution/phases/PhasesController.kt
@@ -1,5 +1,7 @@
 package org.utbot.instrumentation.instrumentation.execution.phases
 
+import com.jetbrains.rd.util.error
+import com.jetbrains.rd.util.getLogger
 import org.utbot.common.StopWatch
 import org.utbot.common.ThreadBasedExecutor
 import org.utbot.framework.plugin.api.Coverage
@@ -17,7 +19,7 @@ import org.utbot.instrumentation.instrumentation.execution.context.Instrumentati
 import java.security.AccessControlException
 
 class PhasesController(
-    instrumentationContext: InstrumentationContext,
+    private val instrumentationContext: InstrumentationContext,
     traceHandler: TraceHandler,
     delegateInstrumentation: Instrumentation<Result<*>>,
     private val timeout: Long
@@ -56,13 +58,27 @@ class PhasesController(
         }
     }
 
+    companion object {
+        private val logger = getLogger<PhasesController>()
+    }
+
     fun <T, R : ExecutionPhase> executePhaseInTimeout(phase: R, block: R.() -> T): T = phase.start {
         val stopWatch = StopWatch()
         val context = UtContext(utContext.classLoader, stopWatch)
         val timeoutForCurrentPhase = timeout - currentlyElapsed
         val result = ThreadBasedExecutor.threadLocal.invokeWithTimeout(timeout - currentlyElapsed, stopWatch) {
             withUtContext(context) {
-                phase.block()
+                try {
+                    phase.block()
+                } catch (outerThrowable: Throwable) {
+                    try {
+                        instrumentationContext.onPhaseUnderTimeoutFailure(phase, outerThrowable)
+                    } catch (innerThrowable: Throwable) {
+                        logger.error("onPhaseUnderTimeoutFailure failed", innerThrowable)
+                    } finally {
+                        throw outerThrowable
+                    }
+                }
             }
         } ?: throw TimeoutException("Timeout $timeoutForCurrentPhase ms for phase ${phase.javaClass.simpleName} elapsed, controller timeout - $timeout")
 

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringInstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringInstrumentationContext.kt
@@ -54,6 +54,6 @@ class SpringInstrumentationContext(
         if (classId.isSubtypeOf(resultActionsClassId)) UtMockMvcResultActionsModelConstructor()
         else delegateInstrumentationContext.findUtModelWithCompositeOriginConstructor(classId)
 
-    override fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable) =
+    override fun onPhaseTimeout(timedOutedPhase: ExecutionPhase) =
         springApi.afterTestMethod()
 }

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringInstrumentationContext.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringInstrumentationContext.kt
@@ -11,6 +11,7 @@ import org.utbot.framework.plugin.api.util.isSubtypeOf
 import org.utbot.framework.plugin.api.util.utContext
 import org.utbot.instrumentation.instrumentation.execution.constructors.UtModelWithCompositeOriginConstructor
 import org.utbot.instrumentation.instrumentation.execution.context.InstrumentationContext
+import org.utbot.instrumentation.instrumentation.execution.phases.ExecutionPhase
 import org.utbot.spring.api.SpringApi
 import org.utbot.spring.api.provider.SpringApiProviderFacade
 import org.utbot.spring.api.provider.InstantiationSettings
@@ -52,4 +53,7 @@ class SpringInstrumentationContext(
     override fun findUtModelWithCompositeOriginConstructor(classId: ClassId): UtModelWithCompositeOriginConstructor? =
         if (classId.isSubtypeOf(resultActionsClassId)) UtMockMvcResultActionsModelConstructor()
         else delegateInstrumentationContext.findUtModelWithCompositeOriginConstructor(classId)
+
+    override fun onPhaseUnderTimeoutFailure(failedPhase: ExecutionPhase, throwable: Throwable) =
+        springApi.afterTestMethod()
 }

--- a/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringUtExecutionInstrumentation.kt
+++ b/utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/instrumentation/spring/SpringUtExecutionInstrumentation.kt
@@ -65,7 +65,7 @@ class SpringUtExecutionInstrumentation(
         arguments: ArgumentList,
         parameters: Any?,
         phasesWrapper: PhasesController.(invokeBasePhases: () -> UtConcreteExecutionResult) -> UtConcreteExecutionResult
-    ): UtConcreteExecutionResult {
+    ): UtConcreteExecutionResult = synchronized(this) {
         getRelevantBeans(clazz).forEach { beanName -> springApi.resetBean(beanName) }
         return delegateInstrumentation.invoke(clazz, methodSignature, arguments, parameters) { invokeBasePhases ->
             phasesWrapper {

--- a/utbot-spring-commons/src/main/kotlin/org/utbot/spring/SpringApiImpl.kt
+++ b/utbot-spring-commons/src/main/kotlin/org/utbot/spring/SpringApiImpl.kt
@@ -152,6 +152,10 @@ class SpringApiImpl(
     }
 
     override fun afterTestMethod() {
+        if (!isInsideTestMethod) {
+            logger.warn { "afterTestMethod() was probably called twice, ignoring second call" }
+            return
+        }
         testContextManager.afterTestExecution(dummyTestClassInstance, dummyTestMethod, null)
         testContextManager.afterTestMethod(dummyTestClassInstance, dummyTestMethod, null)
         isInsideTestMethod = false

